lib/fluent/config/types.rb

```diff
@@ -67,6 +67,14 @@ def self.bool_value(str)
     val
   })
 
+  Configurable.register_type(:enum, Proc.new { |val, opts|
+    s = val.to_sym
+    unless opts[:list].include?(s)
```

Comment:
Maybe, checking `:list` is needed. Current code raises nil related exception without `:list` option.